plugins/modules/ini_file.py

```diff
@@ -46,6 +46,24 @@
       - If left empty, being omitted, or being set to V(null), the O(option) will be placed before the first O(section).
       - Using V(null) is also required if the config format does not support sections.
     type: str
+  section_has:
+    type: list
+    elements: dict
+    required: false
+    suboptions:
+      option:
+        type: str
+        description: The option to look for within the section.
+        required: true
+      value:
+        type: str
+        description: Locate the section with this specific value.
+        required: true
+    description:
+      - Among possibly multiple sections of the same name, select the one that contains these values.
```

Comment:
I feel`section_has` would be easier to understand if its description included something along the lines of "section has these values"

Reply:
Maybe it would also be a good idea to rename the option to `section_has_values`.

Reply:
How do you feel about this idea? I have been using it to set up some Wireguard peers (similar to the examples), and figured it might be a useful addition.

I guess a lot of little adjustments could be made... There's also part of the behaviour wrt "section" that's not terribly clear from the documentation (specifically, if you set "section", leave out "option", and set "state=absent", the entire section is removed from the file). 

I'd also prefer to add a couple of test cases.

Do you think it's worth the time fiddling with that? ... And if yes, is it correct that the place to add test cases would be in `tests/integration/targets/ini_file/tasks/tests`?

Wrt the name of the option ( /section_has/ or /section_has_values/ ), I thought the former to be easy to understand when I wrote my playbook task (which is similar to the example). It's quite clear from the context that it's related to an option and value. And it's shorter, which I like.

Reply:
IMO adding this is OK. Not sure what others think though :)

> I'd also prefer to add a couple of test cases, including some for the existing *section* functionality

:+1: for that!

> And if yes, is it correct that the place to add test cases would be in `tests/integration/targets/ini_file/tasks/tests`?

Yes, that's the right place. If you add a new file there, also remember to add it to tests/integration/targets/ini_file/tasks/main.yml.

(You can run the tests with `ansible-test integration -v ini_file`, assuming you checked out the repository in a directory tree structure that ansible-test needs. ansible-test will tell you if you didn't.)

> Wrt the name of the option (*section_has* or *section_has_values*), I thought the former to be easy to understand when I wrote my playbook task (which is similar to the example). It seems clear from the context, that it's related to an option and value. And it's shorter, which I like.

In the playbook itself `section_has` is sufficient, but just looking at the name without seeing an example, I think the longer name is better. Since the option name shows up in both situations (docs vs. examples/playbooks), I think it is better to use the slightly longer one.

Would be nice to also hear more opinions here, such questions are highly subjective :)

Reply:
@jpmens any opinions on this?

Reply:
Yes, I too prefer the longer name, and that in spite of me typically liking short variable names, etc. :) As already mentioned I think the longer parameter name explains itself better. I thus agree with @felixfontein